components/codec/src/byte.rs

```diff
@@ -372,7 +367,8 @@ impl MemComparableByteCodec {
                     // is faster than checking pad bytes one by one, since it will compare multiple
                     // bytes at once.
                     let base_padding_ptr = dest_ptr.sub(padding_size);
-                    let expected_padding_ptr = T::get_raw_padding_ptr();
+                    // Force a compile time check to ensure safety. The check will be optimized away.
+                    let expected_padding_ptr = T::PADDING[..padding_size].as_ptr();
```

Comment:
padding_size is not a constant but a variable, how would compiler discover that there is out of bound possibility?

Reply:
Because L362 has checked that `padding_size <= MEMCMP_GROUP_SIZE`, and the len of `T::PADDING` is exact `MEMCMP_GROUP_SIZE`. /cc @skyzh

Reply:
This sounds reasonable to me.